### PR TITLE
:sparkles: Add Microsoft CLU NLU

### PR DESCRIPTION
**Pull request by rmtuckerphx**
## Proposed Changes

:sparkles: Add NLU plugin for Microsoft Conversation Language Understanding (CLU)

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by rmtuckerphx** [comment]
@aswetlow @jankoenig When you can, please let me know how soon this can be in a release.

Thank you!

File: integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+export class MicrosoftCluNlu extends NluPlugin<MicrosoftCluNluConfig> {
+  getDefaultConfig(): MicrosoftCluNluConfig {
+    return {
+      ...super.getDefaultConfig(),
+      fallbackLanguage: 'en',
+      endpoint: '',
+      libraryConfig: {
+        id: uuidV4(),
```
Comment (aswetlow) [inline integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: Isn't `id` and `participantId` more like request-related data? The current solution would result in the same ids for every request. 

I can't find more information about that in the CLU docs...

File: integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      });
+    }
+
+    try {
+      const result = await this.sendTextToServer(jovo, text);
+
+      if (
+        result &&
+        result.prediction &&
+        result.prediction.projectKind === PROJECTKIND_CONVERSATION
```
Comment (aswetlow) [inline integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: We mainly use the short form:
`result?.prediction?.projectKind === PROJECTKIND_CONVERSATION`

File: integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        nluData.intent = { name: topIntent };
+        nluData.entities = this.getEntityMapFromResponse(result);
+        nluData.native = result;
+
+        return nluData;
+      }
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error('Error while retrieving nlu-data from MicrosoftClu-server: ', e);
+      return;
```
Comment (aswetlow) [inline integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: Can be removed. Same for the `return` below.

**Review by aswetlow — CHANGES_REQUESTED** [review]
Thank you for the PR, @rmtuckerphx 

I added a few comments/questions

File: integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts (hunk 1 above)
Comment (rmtuckerphx) [reply integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: I couldn't find more info about them either. Here is one of the few samples and they just set the value to "1":
https://learn.microsoft.com/en-us/rest/api/language/2022-05-01/conversation-analysis-runtime/analyze-conversation?tabs=HTTP#conversation-project-result

That's why I surfaced it in config but defaulted to a UUID

File: integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts (hunk 3 above)
Comment (rmtuckerphx) [reply integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: @aswetlow Why can this be removed?

File: integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts (hunk 1 above)
Comment (rmtuckerphx) [reply integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: @aswetlow  Do you have a better idea of how to approach this?

Comment (rmtuckerphx) [reply integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: I posted a question on the Microsoft forum here:
https://learn.microsoft.com/en-us/answers/questions/1276259/when-making-an-api-request-to-clu-analysis-what-ar

Comment (rmtuckerphx) [reply integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts]: @aswetlow Looking at the most recent responses from the Microsoft forum, I think I'm going to remove the config settings for `id` and `participantId` and instead set the `id` value to `request.id` and the `participantId` to the string `"User"`.

What do you think?

**Comment by rmtuckerphx** [comment]
@aswetlow All fixes have been made.

File: integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    if (!this.config.libraryConfig.taskParameters.deploymentName) {
+      throw new JovoError({
+        message: `Can not send request to Microsoft-CLU. The deploymentName is missing.`,
+      });
+    }
+
+    const result = await this.sendTextToServer(jovo, text);
+
+    if (result.prediction.projectKind === PROJECTKIND_CONVERSATION) {
```
Comment (aswetlow) [inline integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts:71]: Is there a case where `result` or `result.prediction` could be undefined?

Comment (rmtuckerphx) [reply integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts:71]: @aswetlow No. Both prediction and projectKind are required and don't support undefined.

```ts
export declare interface AnalyzeConversationResult {
    /** The conversation utterance given by the caller. */
    query: string;
    /** The system detected language for the query in BCP 47 language representation.. */
    detectedLanguage?: string;
    /** The prediction result of a conversation project. */
    prediction: BasePredictionUnion;
}

/** This is the base class of prediction */
export declare interface BasePrediction {
    /** Polymorphic discriminator, which specifies the different types this object can be */
    projectKind: "Conversation" | "Orchestration";
    /** The intent with the highest score. */
    topIntent?: string;
}

export declare type BasePredictionUnion = ConversationPrediction | OrchestrationPrediction;

/** Represents the prediction section of a Conversation project. */
export declare interface ConversationPrediction extends BasePrediction {
    /** Polymorphic discriminator, which specifies the different types this object can be */
    projectKind: "Conversation";
    /** The intent classification results. */
    intents: ConversationIntent[];
    /** The entity extraction results. */
    entities: ConversationEntity[];
}

/** This represents the prediction result of an Orchestration project. */
export declare interface OrchestrationPrediction extends BasePrediction {
    /** Polymorphic discriminator, which specifies the different types this object can be */
    projectKind: "Orchestration";
    /** A dictionary that contains all intents. A key is an intent name and a value is its confidence score and target type. The top intent's value also contains the actual response from the target project. */
    intents: {
        [propertyName: string]: TargetIntentResultUnion;
    };
}
```

**Review by jankoenig — APPROVED** [review]
Looking good. Thank you @rmtuckerphx 🎉

**Review by aswetlow — APPROVED** [review]

**Pull request merged**